src/reanimated2/commonTypes.ts

```diff
@@ -94,30 +94,92 @@ export type MapperRegistry = {
   stop: (mapperID: number) => void;
 };
 
-export type Context = Record<string, unknown>;
+export type WorkletClosure = Record<string, unknown>;
 
-export interface WorkletFunction {
-  _closure?: Context;
+interface ReleaseInitData {
+  code: string;
+}
+
+interface DevInitData {
+  code: string;
+  location: string;
+  sourceMap: string;
+  version: string;
+}
+interface ReleaseWorkletBase {
+  _closure: WorkletClosure;
+  __initData: ReleaseInitData;
+  __workletHash: number;
+}
+
+interface DevWorkletBase {
+  _closure: WorkletClosure;
+  __initData: DevInitData;
+  __workletHash: number;
+  __stackDetails: Error;
+}
+
+// A stands for Arguments, R stands for Return value
+
+type ReleaseWorkletFunction<A extends unknown[], R> = ReleaseWorkletBase & {
+  (...args: A): R;
+};
+
+type DevWorkletFunction<A extends unknown[], R> = DevWorkletBase & {
+  (...args: A): R;
+};
+
+export type WorkletFunction<A extends unknown[], T> =
+  | ReleaseWorkletFunction<A, T>
+  | DevWorkletFunction<A, T>;
+
+/**
+ * @deprecated
+ */
+export type __Context = Record<string, unknown>;
```

Comment:
But we already have `WorkletContext`, right?

Reply:
Yes, but I don't want to mix the new correct types with the old types. Also `__Context` is imported in a few places.